pkg/rehearse/rehearse.go

```diff
@@ -637,6 +653,56 @@ func ensureImageStreamTags(ctx context.Context, client ctrlruntimeclient.Client,
 	return g.Wait()
 }
 
+func ensureISTSInQCI(ctx context.Context, client ctrlruntimeclient.Client,
+	ists apihelper.ImageStreamTagMap, registryConfig string, quayIOImageHelper quayiociimagesdistributor.OCClient, log *logrus.Entry) error {
+
+	mirrorOptions := quayiociimagesdistributor.OCImageMirrorOptions{
+		RegistryConfig:  registryConfig,
+		ContinueOnError: true,
+		MaxPerRegistry:  20,
+		BatchSize:       10,
+		DryRun:          false,
+	}
+	var errs []error
+	for _, ist := range ists {
+
+		colonSplit := strings.Split(ist.Name, ":")
+		if n := len(colonSplit); n != 2 {
+			return fmt.Errorf("splitting %s by `:` didn't yield two but %d results", ist.Name, n)
+		}
+		tagRef := api.ImageStreamTagReference{Namespace: ist.Namespace, Name: colonSplit[0], Tag: colonSplit[1]}
+		quayImage := api.QuayImage(tagRef)
+		sourceImageStreamTag := &imagev1.ImageStreamTag{}
+		if err := client.Get(ctx, ist, sourceImageStreamTag); err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Debug("Source imageStreamTag not found")
+				return nil
+			}
+			return fmt.Errorf("failed to get imageStreamTag %s from registry cluster: %w", ist.String(), err)
+		}
+
+		imageName := sourceImageStreamTag.Image.ObjectMeta.Name
+		colonSplit = strings.Split(imageName, ":")
+		if n := len(colonSplit); n != 2 {
+			//should never happen
+			return fmt.Errorf("splitting %s by `:` didn't yield two but %d results", imageName, n)
+		}
+		if colonSplit[0] != "sha256" {
+			//should never happen
+			return fmt.Errorf("image name has no prefix `sha256:`: %s", imageName)
+		}
+
+		sourceImage := fmt.Sprintf("%s/%s/%s@%s", api.DomainForService(api.ServiceRegistry), tagRef.Namespace, tagRef.Name, sourceImageStreamTag.Image.ObjectMeta.Name)
+		targetImageWithDateAndDigest := api.QuayImageFromDateAndDigest(time.Now().Format("20060102"), colonSplit[1])
+
+		pairs := []string{sourceImage + "=" + quayImage, sourceImage + "=" + targetImageWithDateAndDigest}
+		if err := quayIOImageHelper.ImageMirror(pairs, mirrorOptions); err != nil {
```

Comment:
Could we dry out a function to compute `pairs`?
I would like to see some unit tests on it.